File: src/test/java/com/morak/performancetracker/DatabaseInitializationTest.java
package com.morak.performancetracker;

import static org.assertj.core.api.Assertions.assertThat;

import com.morak.performancetracker.aop.persistence.QueryMonitor;
import org.junit.jupiter.api.Test;
import org.springframework.beans.factory.annotation.Autowired;
import org.springframework.boot.test.context.SpringBootTest;
import org.springframework.test.context.jdbc.Sql;

@SpringBootTest
@PerformanceTracker
@Sql("classpath:schema.sql")
public class DatabaseInitializationTest {

    private final QueryMonitor queryMonitor;

    @Autowired
    public DatabaseInitializationTest(QueryMonitor queryMonitor) {
        this.queryMonitor = queryMonitor;
    }

    @Test
    void SQL파일에_의한_쿼리는_측정하지_않는다() {
        assertThat(queryMonitor.getQueryCount()).isZero();
    }
}

File: src/test/kotlin/com/morak/performancetracker/DatabaseInitializationTest.kt
package com.morak.performancetracker

import com.morak.performancetracker.aop.persistence.QueryMonitor
import org.assertj.core.api.Assertions.assertThat
import org.junit.jupiter.api.Test
import org.springframework.boot.test.context.SpringBootTest
import org.springframework.test.context.TestConstructor
import org.springframework.test.context.jdbc.Sql

@SpringBootTest
@TestConstructor(autowireMode = TestConstructor.AutowireMode.ALL)
@PerformanceTracker
@Sql("classpath:schema.sql")
class DatabaseInitializationTest(private val queryMonitor: QueryMonitor) {
    @Test
    fun `SQL파일에 의한 쿼리는 측정하지 않는다`() {
        assertThat(queryMonitor.query).isNull()
    }
}
